Framework/DataHandling/src/LoadSampleEnvironment.cpp

```diff
@@ -0,0 +1,118 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidDataHandling/LoadSampleEnvironment.h"
+#include "MantidDataHandling/LoadAsciiStl.h"
+#include "MantidDataHandling/LoadBinaryStl.h"
+#include "MantidGeometry/Instrument/Container.h"
+#include "MantidGeometry/Instrument/SampleEnvironment.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Sample.h"
+
+#include "MantidKernel/Exception.h"
+
+#include <Poco/File.h>
+#include <boost/algorithm/string.hpp>
+#include <fstream>
+#include <stdio.h>
+#include <string.h>
+
+namespace Mantid {
+namespace DataHandling {
+
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(LoadSampleEnvironment)
+
+using namespace Kernel;
+using namespace API;
+using namespace Geometry;
+
+void LoadSampleEnvironment::init() {
+  auto wsValidator = boost::make_shared<API::InstrumentValidator>();
+  ;
+
+  // input workspace
+  declareProperty(make_unique<WorkspaceProperty<>>(
+                      "InputWorkspace", "", Direction::Input, wsValidator),
+                  "The name of the workspace containing the instrument to add "
+                  "the Environment");
+
+  // Environment file
+  const std::vector<std::string> extensions{".stl"};
+  declareProperty(
+      make_unique<FileProperty>("Filename", "", FileProperty::Load, extensions),
+      "The path name of the file containing the Environment");
+
+  // Output workspace
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "The name of the workspace that will contain the loaded "
+                  "Environment of the sample");
+
+  // Environment Name
+  declareProperty("EnvironmentName", "Environment");
+
+  // New Can or Add
+  declareProperty("Add", false);
```

Comment:
I think the concept of adding to an existing environment needs further explanation. Also what does "Can" mean in your comment above?

Reply:
So, an environment is made up of the can, and then a vector of additional parts (I assume it was implemented this way since each section of the environment can have a different material) Since STL files can't hold information on things like material, the option for adding to an existing environment will allow the user to create a complex environment of multiple materials, which will be needed for absorption corrections